{"topicSections":[{"identifiers":["doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError\/init(from:)"],"anchor":"Initializers","generated":true,"title":"Initializers"},{"identifiers":["doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError\/error"],"anchor":"Instance-Properties","generated":true,"title":"Instance Properties"}],"metadata":{"roleHeading":"Structure","navigatorTitle":[{"text":"APIError","kind":"identifier"}],"modules":[{"name":"OpenAIKit"}],"symbolKind":"struct","role":"symbol","title":"APIError","externalID":"s:9OpenAIKit8APIErrorV","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"text":"APIError","kind":"identifier"}]},"abstract":[{"text":"A structured error response from the OpenAI API.","type":"text"}],"variants":[{"paths":["\/documentation\/openaikit\/apierror"],"traits":[{"interfaceLanguage":"swift"}]}],"kind":"symbol","seeAlsoSections":[{"identifiers":["doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError"],"anchor":"Error-Handling","generated":true,"title":"Error Handling"}],"relationshipsSections":[{"identifiers":["doc:\/\/OpenAIKit\/Se","doc:\/\/OpenAIKit\/SE","doc:\/\/OpenAIKit\/s8SendableP"],"type":"conformsTo","kind":"relationships","title":"Conforms To"}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError"},"schemaVersion":{"major":0,"patch":0,"minor":3},"hierarchy":{"paths":[["doc:\/\/OpenAIKit\/documentation\/OpenAIKit"]]},"sections":[],"primaryContentSections":[{"kind":"declarations","declarations":[{"languages":["swift"],"tokens":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"APIError"}],"platforms":["Linux"]}]},{"content":[{"anchor":"overview","type":"heading","level":2,"text":"Overview"},{"inlineContent":[{"text":"When the OpenAI API encounters an error, it returns a structured JSON response","type":"text"},{"text":" ","type":"text"},{"text":"containing detailed information about what went wrong. This struct represents","type":"text"},{"text":" ","type":"text"},{"text":"the top-level error container.","type":"text"}],"type":"paragraph"},{"anchor":"Structure","type":"heading","level":2,"text":"Structure"},{"inlineContent":[{"text":"The API returns errors in the following format:","type":"text"}],"type":"paragraph"},{"code":["{","  \"error\": {","    \"message\": \"Invalid API key provided\",","    \"type\": \"invalid_request_error\",","    \"param\": null,","    \"code\": \"invalid_api_key\"","  }","}"],"syntax":"json","type":"codeListing"},{"anchor":"Usage","type":"heading","level":2,"text":"Usage"},{"code":["do {","    let completion = try await openAI.completions.create(...)","} catch OpenAIError.apiError(let apiError) {","    \/\/ Access the detailed error information","    let errorDetail = apiError.error","    print(\"Error: \\(errorDetail.message)\")","    ","    if let errorType = errorDetail.type {","        print(\"Type: \\(errorType)\")","    }","}"],"syntax":"swift","type":"codeListing"},{"content":[{"type":"paragraph","inlineContent":[{"text":"","type":"text"},{"isActive":true,"identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIErrorDetail","type":"reference"},{"text":", ","type":"text"},{"isActive":true,"identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError\/apiError(_:)","type":"reference"}]}],"type":"aside","style":"note","name":"See Also"}],"kind":"content"}],"references":{"doc://OpenAIKit/s8SendableP":{"title":"Swift.Sendable","identifier":"doc:\/\/OpenAIKit\/s8SendableP","type":"unresolvable"},"doc://OpenAIKit/documentation/OpenAIKit/APIError/init(from:)":{"url":"\/documentation\/openaikit\/apierror\/init(from:)","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError\/init(from:)","title":"init(from:)","role":"symbol","fragments":[{"kind":"identifier","text":"init"},{"kind":"text","text":"("},{"kind":"externalParam","text":"from"},{"kind":"text","text":": any "},{"preciseIdentifier":"s:s7DecoderP","kind":"typeIdentifier","text":"Decoder"},{"kind":"text","text":") "},{"kind":"keyword","text":"throws"}],"abstract":[],"kind":"symbol","type":"topic"},"doc://OpenAIKit/documentation/OpenAIKit":{"url":"\/documentation\/openaikit","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit","title":"OpenAIKit","role":"collection","abstract":[{"text":"A powerful, type-safe Swift SDK for the OpenAI API with support for all major endpoints and platforms.","type":"text"}],"kind":"symbol","type":"topic"},"doc://OpenAIKit/documentation/OpenAIKit/OpenAIError/apiError(_:)":{"url":"\/documentation\/openaikit\/openaierror\/apierror(_:)","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError\/apiError(_:)","title":"OpenAIError.apiError(_:)","role":"symbol","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"apiError"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"APIError","preciseIdentifier":"s:9OpenAIKit8APIErrorV"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"An error response was received from the OpenAI API."}],"kind":"symbol","type":"topic"},"doc://OpenAIKit/documentation/OpenAIKit/APIError/error":{"type":"topic","abstract":[{"text":"The detailed error information from the API.","type":"text"}],"fragments":[{"kind":"keyword","text":"let"},{"kind":"text","text":" "},{"kind":"identifier","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"APIErrorDetail","preciseIdentifier":"s:9OpenAIKit14APIErrorDetailV"}],"identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError\/error","role":"symbol","url":"\/documentation\/openaikit\/apierror\/error","title":"error","kind":"symbol"},"doc://OpenAIKit/documentation/OpenAIKit/APIError":{"identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError","type":"topic","role":"symbol","title":"APIError","abstract":[{"text":"A structured error response from the OpenAI API.","type":"text"}],"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"APIError"}],"url":"\/documentation\/openaikit\/apierror","navigatorTitle":[{"kind":"identifier","text":"APIError"}],"kind":"symbol"},"doc://OpenAIKit/documentation/OpenAIKit/APIErrorDetail":{"url":"\/documentation\/openaikit\/apierrordetail","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIErrorDetail","title":"APIErrorDetail","role":"symbol","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"APIErrorDetail","kind":"identifier"}],"abstract":[{"text":"Detailed error information from the OpenAI API.","type":"text"}],"kind":"symbol","type":"topic","navigatorTitle":[{"text":"APIErrorDetail","kind":"identifier"}]},"doc://OpenAIKit/Se":{"type":"unresolvable","identifier":"doc:\/\/OpenAIKit\/Se","title":"Swift.Decodable"},"doc://OpenAIKit/documentation/OpenAIKit/OpenAIError":{"navigatorTitle":[{"text":"OpenAIError","kind":"identifier"}],"title":"OpenAIError","abstract":[{"type":"text","text":"The primary error type for all OpenAI API interactions."}],"role":"symbol","kind":"symbol","type":"topic","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError","url":"\/documentation\/openaikit\/openaierror","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"OpenAIError","kind":"identifier"}]},"doc://OpenAIKit/SE":{"identifier":"doc:\/\/OpenAIKit\/SE","type":"unresolvable","title":"Swift.Encodable"}}}